source/mqtt/MqttClient.cpp

```diff
@@ -303,7 +338,7 @@ namespace Aws
             {
                 auto connection = reinterpret_cast<MqttConnection *>(user_data);
 
-                Allocator *allocator = connection->m_owningClient->allocator;
+                Allocator *allocator = connection->m_allocator;
```

Comment:
`m_owningClient` is a mqtt3 client. As now the `m_owningClient` is not necessary set for the case of Mqtt5, use the m_allocator of connection instead of using the allocator from mqtt client.